arch/x86/core/intel64/cpu.c

```diff
@@ -15,133 +17,37 @@
-	.iomapb = 0xFFFF,
-	.cpu = &(_kernel.cpus[0])
-};
-
-#if CONFIG_MP_MAX_NUM_CPUS > 1
-Z_GENERIC_SECTION(.tss)
-struct x86_tss64 tss1 = {
-#ifdef CONFIG_X86_KPTI
-	.ist2 = (uint64_t) z_x86_trampoline_stack1 + Z_X86_TRAMPOLINE_STACK_SIZE,
-#endif
-	.ist6 = (uint64_t) z_x86_nmi_stack1 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.ist7 = (uint64_t) z_x86_exception_stack1 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.iomapb = 0xFFFF,
-	.cpu = &(_kernel.cpus[1])
-};
-#endif
-
-#if CONFIG_MP_MAX_NUM_CPUS > 2
-Z_GENERIC_SECTION(.tss)
-struct x86_tss64 tss2 = {
-#ifdef CONFIG_X86_KPTI
-	.ist2 = (uint64_t) z_x86_trampoline_stack2 + Z_X86_TRAMPOLINE_STACK_SIZE,
-#endif
-	.ist6 = (uint64_t) z_x86_nmi_stack2 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.ist7 = (uint64_t) z_x86_exception_stack2 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.iomapb = 0xFFFF,
-	.cpu = &(_kernel.cpus[2])
-};
-#endif
-
-#if CONFIG_MP_MAX_NUM_CPUS > 3
-Z_GENERIC_SECTION(.tss)
-struct x86_tss64 tss3 = {
-#ifdef CONFIG_X86_KPTI
-	.ist2 = (uint64_t) z_x86_trampoline_stack3 + Z_X86_TRAMPOLINE_STACK_SIZE,
-#endif
-	.ist6 = (uint64_t) z_x86_nmi_stack3 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.ist7 = (uint64_t) z_x86_exception_stack3 + CONFIG_X86_EXCEPTION_STACK_SIZE,
-	.iomapb = 0xFFFF,
-	.cpu = &(_kernel.cpus[3])
-};
+#if !defined(CONFIG_ACPI)
+#define INIT_CPUID(n, _) n
+__weak uint8_t x86_cpu_loapics[] = {
+	LISTIFY(CONFIG_MP_MAX_NUM_CPUS, INIT_CPUID, (,)),};
+#else
+__weak uint8_t x86_cpu_loapics[CONFIG_MP_MAX_NUM_CPUS];
 #endif
+static struct x86_cpu_info *cpu_info[CONFIG_MP_MAX_NUM_CPUS];
+extern char x86_ap_start[]; /* AP entry point in locore.S */
 
+LISTIFY(CONFIG_MP_MAX_NUM_CPUS, ACPI_CPU_INIT, (;));
 
-/* We must put this in a dedicated section, or else it will land into .bss:
- * in this case, though locore.S initalizes it relevantly, all will be
- * lost when calling z_bss_zero() in z_x86_cpu_init prior to using it.
- */
 Z_GENERIC_SECTION(.boot_arg)
 x86_boot_arg_t x86_cpu_boot_arg;
 
 struct x86_cpuboot x86_cpuboot[] = {
-	{
-		.tr = X86_KERNEL_CPU0_TR,
-		.gs_base = &tss0,
-		.sp = (uint64_t) z_interrupt_stacks[0] +
-			Z_KERNEL_STACK_SIZE_ADJUST(CONFIG_ISR_STACK_SIZE),
-		.stack_size =
-			Z_KERNEL_STACK_SIZE_ADJUST(CONFIG_ISR_STACK_SIZE),
-		.fn = z_x86_prep_c,
-		.arg = &x86_cpu_boot_arg,
-	},
-#if CONFIG_MP_MAX_NUM_CPUS > 1
-	{
-		.tr = X86_KERNEL_CPU1_TR,
-		.gs_base = &tss1
-	},
-#endif
-#if CONFIG_MP_MAX_NUM_CPUS > 2
-	{
-		.tr = X86_KERNEL_CPU2_TR,
-		.gs_base = &tss2
-	},
-#endif
-#if CONFIG_MP_MAX_NUM_CPUS > 3
-	{
-		.tr = X86_KERNEL_CPU3_TR,
-		.gs_base = &tss3
-	},
-#endif
+	LISTIFY(CONFIG_MP_MAX_NUM_CPUS, X86_CPU_BOOT_INIT, (,)),
 };
 
-/*
- * Send the INIT/STARTUP IPI sequence required to start up CPU 'cpu_num', which
- * will enter the kernel at fn(arg), running on the specified stack.
- */
-
-void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
-		    arch_cpustart_t fn, void *arg)
+void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz, arch_cpustart_t fn, void *arg)
```

Comment:
Please do not mix formatting changes into commits containing functional changes, which makes it harder to review the functional changes. Ditto for formatting changes below.